File: SPSP/SPSP.Models/Report/MenuItemReportData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SPSP.Models
{
    public class MenuItemReportData
    {
        public string Name { get; set; }    
        public string Category { get; set; }
        public decimal Price { get; set; }
        public int OrderCount { get; set; }
        public decimal TotalAmount { get; set; }    
    }
}

File: SPSP/UI/spsp_desktop/lib/models/menu_item_report_data/menu_item_report_data.dart
import 'package:json_annotation/json_annotation.dart';

part 'menu_item_report_data.g.dart';

@JsonSerializable()
class MenuItemReportData {
  int? id;
  String? name;
  String? category;
  double price;
  int orderCount;
  double totalAmount;

  MenuItemReportData({
    this.id,
    this.name,
    this.category,
    required this.price,
    required this.orderCount,
    required this.totalAmount,
  });

  factory MenuItemReportData.fromJson(Map<String, dynamic> json) =>
      _$MenuItemReportDataFromJson(json);

  Map<String, dynamic> toJson() => _$MenuItemReportDataToJson(this);
}
